src/gui/dialogs/file_dialog.cpp

```diff
@@ -196,8 +196,14 @@ void file_dialog::check_filename() {
 			}
 		}
 
+		utils::string_map i18n_strings;
+		i18n_strings["red_x_mark"] = "<span color='red' face='DejaVuSans'>✘</span>";
 		if (wrong_ext) {
-			validation_msg.set_label("<span color='red' face='DejaVuSans'>✘</span><span color='red' size='small'>wrong extension, use " + utils::join(extensions_, ", ") + "</span>");
+			i18n_strings["extensions"] = utils::join(extensions_, ", ");
+			validation_msg.set_label(VGETTEXT("$red_x_mark <span color='red' size='small'>wrong extension, use $extensions</span>", i18n_strings));
+			save_btn.set_active(false);
+		} else if (filename.find(' ') != std::string::npos) {
+			validation_msg.set_label(VGETTEXT("$red_x_mark <span color='red' size='small'>whitespace is not allowed</span>", i18n_strings));
```

Comment:
This only checks for space but whitespace sounds like it also checks for tab etc.

Is it clear to the user that this is about the file name containing a space?

Do we disallow spaces in any file name? Savegames definitely do contain spaces for example. Not sure if this dialog is used for savegames though.

Reply:
updated check and message, see https://github.com/wesnoth/wesnoth/pull/9273/commits/0b4de723d64b3f923a1d8a420ca451623552a7b7.
The file dialog doesn't seem to be used outside of the editor.

Reply:
Alright, good enough for me.

Reply:
> The file dialog doesn't seem to be used outside of the editor.

It's definitely used outside of the editor. I'm not aware of any other place where it's used for _saving_, but at minimum it's used to select a wesnothd executable if the game can't find one in normal locations. If there are any other uses outside the editor, they're about as niche as that one.